### PR TITLE
Nfdiv 384/draft state

### PR DESCRIPTION
**Pull request by velimd**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/NFDIV-384

https://tools.hmcts.net/jira/browse/NFDIV-367

### Change description ###
- added state draft
- included the authorised users
- added new event draftCreate
- included the authorised users for new event
- added field divorceOrDissolution to the new event.

Note: i branched off a branch which was looking into NFDIV-367

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by linusnorton — APPROVED** [review]

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json [hunk 1]
```diff
@@ -7014,5 +7014,47 @@
...
+    "CaseTypeID": "NO_FAULT_DIVORCE",
+    "CaseEventID": "draftCreate",
+    "UserRole": "citizen",
+    "CRUD": "CRU"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "NO_FAULT_DIVORCE",
+    "CaseEventID": "draftCreate",
+    "UserRole": "caseworker-divorce-solicitor",
```
Comment (SarahLittlejohn) [inline definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json:7043]: Unsure if you want all these users to be able to see the create draft event - might just citizens to have CRU and superusers to have R?

File: definitions/divorce/json/AuthorisationCaseField/AuthorisationCaseField.json [hunk 2]
```diff
@@ -19754,5 +19754,47 @@
...
+    "CaseTypeID": "NO_FAULT_DIVORCE",
+    "CaseFieldID": "divorceOrDissolution",
+    "UserRole": "caseworker-divorce-superuser",
+    "CRUD": "R"
+  },
+  {
+    "LiveFrom": "19/01/2021",
+    "CaseTypeID": "NO_FAULT_DIVORCE",
+    "CaseFieldID": "divorceOrDissolution",
+    "UserRole": "caseworker-divorce-courtadmin",
```
Comment (SarahLittlejohn) [inline definitions/divorce/json/AuthorisationCaseField/AuthorisationCaseField.json:19776]: Caseworkers might want CRU rights too

File: definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState.json [hunk 3]
```diff
@@ -2637,5 +2637,68 @@
...
+    "CaseStateID": "Draft",
+    "UserRole": "citizen",
+    "CRUD": "CRU"
+  },
+  {
+    "LiveFrom": "29/01/2021",
+    "CaseTypeID": "NO_FAULT_DIVORCE",
+    "CaseStateID": "Draft",
+    "UserRole": "caseworker-divorce-pcqextractor",
+    "CRUD": "R"
```
Comment (SarahLittlejohn) [inline definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState.json:2702]: Again might just want superusers (R) and citizens (CRU) to have access (potentially pcq-extractor at some point)

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json (hunk 1 above)
Comment (velimd) [reply definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json:7043]: fair, wasnt sure which user had which authentication hence i gave all of them R except for citizen.

File: definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState.json (hunk 3 above)
Comment (velimd) [reply definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState.json:2702]: had to undo this as test failed, seems like you need all 9 users.

**Pull request closed without merging**